packages/mui-joy/src/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,399 @@
+import useSlot from '../utils/useSlot';
+import drawerClasses, { getDrawerUtilityClass } from './drawerClasses';
+import { DrawerProps, DrawerOwnerState, DrawerTypeMap } from './DrawerProps';
+
+const useUtilityClasses = (ownerState: DrawerOwnerState) => {
+  const { open, variant, color, size } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      !open && 'hidden',
+      variant && `variant${capitalize(variant)}`,
+      color && `color${capitalize(color)}`,
+      size && `size${capitalize(size)}`,
+    ],
+    backdrop: ['backdrop'],
+    content: ['content'],
+  };
+
+  return composeClasses(slots, getDrawerUtilityClass, {});
+};
+
+const DrawerRoot = styled('div', {
+  name: 'JoyDrawer',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: DrawerOwnerState }>(({ theme, ownerState }) => ({
+  '--unstable_popup-zIndex': `calc(${theme.vars.zIndex.modal} + 1)`,
+  '& ~ [role="listbox"]': {
+    // target all the listbox (Autocomplete, Menu, Select, etc.) that uses portal
+    '--unstable_popup-zIndex': `calc(${theme.vars.zIndex.modal} + 1)`,
+  },
+  position: 'fixed',
+  zIndex: theme.vars.zIndex.modal,
+  transitionProperty: 'visibility',
+  transitionDelay: ownerState.open ? '0s' : '300ms',
+  ...(!ownerState.open && {
+    visibility: 'hidden',
+  }),
+  [`& .${drawerClasses.backdrop}`]: {
+    opacity: ownerState.open ? 1 : 0,
+    transition: 'opacity 0.3s ease',
+  },
+}));
+
+const DrawerBackdrop = styled(ModalBackdrop as unknown as 'div', {
+  name: 'JoyDrawer',
+  slot: 'Backdrop',
+  overridesResolver: (props, styles) => styles.backdrop,
+})({});
+
+export const DrawerContent = styled(SheetRoot as unknown as 'div', {
+  name: 'JoyDrawer',
+  slot: 'Content',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: DrawerOwnerState }>(({ theme, ownerState }) => ({
+  '--DrawerContent-radius': 0,
+  ...(ownerState.size === 'sm' && {
+    '--DrawerContent-padding': theme.spacing(2),
+    '--DrawerContent-gap': theme.spacing(0.75),
+    '--DrawerContent-titleOffset': theme.spacing(0.25),
+    '--DrawerContent-descriptionOffset': theme.spacing(0.25),
+    '--ModalClose-inset': theme.spacing(1.25),
+  }),
+  ...(ownerState.size === 'md' && {
+    '--DrawerContent-padding': theme.spacing(2.5),
+    '--DrawerContent-gap': theme.spacing(1.5),
+    '--DrawerContent-titleOffset': theme.spacing(0.25),
+    '--DrawerContent-descriptionOffset': theme.spacing(0.75),
+    '--ModalClose-inset': theme.spacing(1.5),
+  }),
+  ...(ownerState.size === 'lg' && {
+    '--DrawerContent-padding': theme.spacing(3),
+    '--DrawerContent-gap': theme.spacing(2),
+    '--DrawerContent-titleOffset': theme.spacing(0.5),
+    '--DrawerContent-descriptionOffset': theme.spacing(1),
+    '--ModalClose-inset': theme.spacing(1.5),
+  }),
+  boxShadow: theme.shadow.md,
+  borderRadius: 'var(--DrawerContent-radius)',
+  fontFamily: theme.vars.fontFamily.body,
+  lineHeight: theme.vars.lineHeight.md,
```

Comment:
Should it be:

```suggestion
  …theme.typography[`body-${ownerState.size}`],
```

other components are using this pattern.